weave/index/index_sql.py

```diff
@@ -0,0 +1,869 @@
+        )
+        # If the database does not exist, create it.
+        if not self.cur.fetchone():
+            self.con.autocommit = True
+            self.cur.execute(f"CREATE DATABASE {database_name};")
+            self.cur.execute(f"USE {database_name};")
+            self.con.autocommit = False
+        else:
+            self.cur.execute(f"USE {database_name};")
+
+    def _create_schema(self):
+        """Create the schema if it does not already exist."""
+        # Check if self.pantry_schema exists.
+        self.cur.execute(
+            "SELECT * FROM sys.schemas WHERE name = ?", (self.pantry_schema,)
+        )
+        # If it does not exist, create it.
+        if not self.cur.fetchone():
+            self.cur.execute(f"CREATE SCHEMA {self.pantry_schema};")
+            self.con.commit()
+
+    def _create_tables(self):
+        """Create the required tables if they do not already exist."""
+        # THIS NEEDS TO BE UPDATED MANUALLY IF NEW COLUMNS ARE ADDED TO INDEX.
+        # THE INSERT IN OTHER FUNCTIONS USE config.index_schema(), BUT THAT
+        # CAN'T BE USED HERE AS TYPE NEEDS TO BE SPECIFIED.
+        self.cur.execute(
+            f"""
+            IF NOT EXISTS (
+                SELECT * FROM sys.tables t
+                JOIN sys.schemas s ON (t.schema_id = s.schema_id)
+                WHERE s.name = '{self.pantry_schema}'
+                AND t.name = 'pantry_index'
+            )
+            CREATE TABLE {self.pantry_schema}.pantry_index (
+                uuid varchar(64),
+                upload_time INT,
+                parent_uuids TEXT,
+                basket_type TEXT,
+                label TEXT,
+                weave_version TEXT,
+                address TEXT,
+                storage_type TEXT,
+                PRIMARY KEY(uuid),
+                UNIQUE(uuid)
+            );
+            """
+        )
+
+        self.cur.execute(
+            f"""
+            IF NOT EXISTS (
+                SELECT * FROM sys.tables t
+                JOIN sys.schemas s ON (t.schema_id = s.schema_id)
+                WHERE s.name = '{self.pantry_schema}'
+                AND t.name = 'parent_uuids'
+            )
+            CREATE TABLE {self.pantry_schema}.parent_uuids (
+                uuid varchar(64),
+                parent_uuid varchar(64),
+                PRIMARY KEY(uuid, parent_uuid),
+                UNIQUE(uuid, parent_uuid)
+            );
+            """
+        )
+        self.con.commit()
+
+    @property
+    def file_system(self):
+        """The file system of the pantry referenced by this Index."""
+        return self._file_system
+
+    @property
+    def pantry_path(self):
+        """The pantry path referenced by this Index."""
+        return self._pantry_path
+
+    def generate_metadata(self, **kwargs):
+        """Populates the metadata for the index.
+
+        Parameters
+        ----------
+        **kwargs unused for this function.
+        """
```

Comment:
I don't want to be nitpicky and won't block for this, but should we have a "returns" section in the docstring? just explaining that we're returning the metadata in a dictionary format?